components/AppBar/src/MDCAppBarNavigationController.m

```diff
@@ -102,23 +103,78 @@ - (void)setViewControllers:(NSArray<UIViewController *> *)viewControllers animat
 }
 
 - (void)setNavigationBarHidden:(BOOL)navigationBarHidden {
-  // TODO: Consider using this API to hide the top view controller's flexible header.
-  NSAssert(navigationBarHidden, @"%@ requires that the system navigation bar remain hidden.",
-           NSStringFromClass([self class]));
+  if (!self.shouldSetNavigationBarHiddenHideAppBar) {
+    NSAssert(navigationBarHidden, @"%@ requires that the system navigation bar remain hidden.",
+             NSStringFromClass([self class]));
+    [super setNavigationBarHidden:YES];
+    return;
+  }
 
   [super setNavigationBarHidden:YES];
+  [self appbar_setNavigationBarHidden:navigationBarHidden animated:NO];
 }
 
 - (void)setNavigationBarHidden:(BOOL)navigationBarHidden animated:(BOOL)animated {
-  // TODO: Consider using this API to hide the top view controller's flexible header.
-  NSAssert(navigationBarHidden, @"%@ requires that the system navigation bar remain hidden.",
-           NSStringFromClass([self class]));
+  if (!self.shouldSetNavigationBarHiddenHideAppBar) {
+    NSAssert(navigationBarHidden, @"%@ requires that the system navigation bar remain hidden.",
+             NSStringFromClass([self class]));
+    [super setNavigationBarHidden:YES animated:animated];
+    return;
+  }
 
   [super setNavigationBarHidden:YES animated:animated];
+  [self appbar_setNavigationBarHidden:navigationBarHidden animated:animated];
+}
+
+- (BOOL)isNavigationBarHidden {
+  if (!self.shouldSetNavigationBarHiddenHideAppBar) {
+    return [super isNavigationBarHidden];
+  }
+
+  MDCAppBarViewController *appBarViewController =
+      [self appBarViewControllerForViewController:self.visibleViewController];
+  if (!appBarViewController) {
+    return YES;
+  }
+  return appBarViewController.headerView.shiftedOffscreen;
 }
 
 #pragma mark - Private
 
+- (void)appbar_setNavigationBarHidden:(BOOL)navigationBarHidden animated:(BOOL)animated {
+  [self appbar_setNavigationBarHidden:navigationBarHidden
+                             animated:animated
+                    forViewController:self.visibleViewController];
+}
+
+- (void)appbar_setNavigationBarHidden:(BOOL)navigationBarHidden
+                             animated:(BOOL)animated
+                    forViewController:(UIViewController *)viewController {
+  MDCAppBarViewController *appBarViewController =
+      [self appBarViewControllerForViewController:viewController];
+  if (!appBarViewController) {
+    return;
+  }
+
+  if (appBarViewController.headerView.shiftBehavior == MDCFlexibleHeaderShiftBehaviorDisabled) {
+    appBarViewController.headerView.shiftBehavior = MDCFlexibleHeaderShiftBehaviorHideable;
```

Comment:
doesn't this depend on the value of navigationBarHidden?

Reply:
In what way do you see it depending on the value? Please expand on what you see that may be missing here.

Reply:
I don't fully understand why the shiftBehavior is changing here regardless of what value the setter received, hence me making sure this wasn't missed. If this isn't affected by the setter value, then does this block of code need to be part of the setter?

Reply:
> If this isn't affected by the setter value

Can you clarify what "this" and "the setter value" means in this context?

Does "this" refer to this line of assignment?

And does "the setter value" refer to the fact that we're assigning `MDCFlexibleHeaderShiftBehaviorHideable` here?

Reply:
While waiting for clarification, the english version of this code is as follows:

"If the shift behavior is presently disabled (the default), then adjust it to be hideable instead, otherwise we do not make any adjustments to the shift behavior because it's likely that the shift behavior was explicitly set to something else."

Reply:
Added this explanation as docs to the code.

Reply:
this = appBarViewController.headerView.shiftBehavior = MDCFlexibleHeaderShiftBehaviorHideable;

setter value = navigationBarHidden

Reply:
Ah I see now. Yes we're changing the behavior of the app bar the moment these APIs are used so that the behavior is instantly "locked in" to the correct "hideable" behavior of the flexible header.